trin-execution/src/evm/mod.rs

```diff
@@ -1,3 +1,61 @@
+use ethportal_api::Header;
+use revm::{inspector_handle_register, inspectors::TracerEip3155, Database, Evm};
+use revm_primitives::{BlobExcessGasAndPrice, BlockEnv, U256};
+use spec_id::get_spec_id;
+use tx_env_modifier::TxEnvModifier;
+
+pub mod async_db;
 pub mod block_executor;
-pub mod blocking;
+pub mod dao_fork;
 pub mod post_block_beneficiaries;
+pub mod spec_id;
+pub mod tx_env_modifier;
+
+/// Creates [BlockEnv] based on data in the [Header].
+pub fn create_block_env(header: &Header) -> BlockEnv {
+    // EIP-4844 excess blob gas of this block, introduced in Cancun
+    let blob_excess_gas_and_price = header
+        .excess_blob_gas
+        .map(|excess_blob_gas| BlobExcessGasAndPrice::new(excess_blob_gas.to()));
+
+    BlockEnv {
+        number: U256::from(header.number),
+        coinbase: header.author,
+        timestamp: U256::from(header.timestamp),
+        gas_limit: header.gas_limit,
+        basefee: header.base_fee_per_gas.unwrap_or_default(),
+        difficulty: header.difficulty,
+        prevrandao: header.mix_hash,
```

Comment:
`prevrandao: header.mix_hash,` I believe we should still be checking if we are past the Merge for `prevrandao`.

mix_hash won't be none in pre-merge blocks, it was used in PoW. Maybe it will be fine? but I would rather not take chances

For difficulty I think what you did is fine as the processed header should always have the right difficulty

Reply:
I didn't know that `header.mix_hash` won't be none pre-merge. I fixed it.